redaxo/src/core/lib/util/stopwatch.php

```diff
@@ -55,6 +55,18 @@ public function stop()
 
     public static function measure($label, callable $callable)
     {
+        static $enabled = false;
+
+        // we might get called very early in the process, in which case we can't determine yet whether the user is logged in.
+        // this also means, in debug-mode we get more timings in comparison to admin-only timings.
+        if (!$enabled) {
+            $enabled = rex::isDebugMode() || ($user = rex::getUser()) && $user->isAdmin();
```

Comment:
Falls der Admin-Check auch im Frontend zuverlässig funktionieren soll, muss man `rex_backend_login::createUser()` statt `rex::getUser()` verwenden, wie zum Beispiel [hier](https://github.com/redaxo/redaxo/blob/2d62d585a3d186dfed33daf5c395903bb98581f3/redaxo/src/core/lib/var_dumper.php#L23)

Reply:
`rex::getUser()` liefert im Frontend nur was, wenn eine andere Stelle zuvor schon `rex_backend_login::createUser()` aufgerufen hat.

Reply:
ich hatte bewusst nicht `createUser` genommen, da man dann eine session brauch und ggf. dann auf einmal session locks in requests hat, die vorher gar keine session gebraucht haben

Reply:
ich bin jetzt quasi davon ausgegangen dass man - wenn man alles sehen will was messbar ist - den debug mode aktivieren muss

Reply:
ok

Reply:
kommentar ergänzt